### DISCUSSION TITLE
about "conformance test"

### DISCUSSION THREAD

**Discussion by choojunyup** [webauthn4j/webauthn4j#550] [Q&A] [answered]
hello.
I've looked through the entire list of issues and couldn't find any information, so I'm asking.
I made a fido server with webauthn4j lib.
And the fido conformance test(1.5.2v) is in progress.
Here are question.

1. How do you set up the "WebAuthnManager" for testing?
    Should I enable both  CertPathTrustworthinessValidator() and fidoMdsMetadataValidator() ?

<pre><code>
CustomWebAuthnConfig(){
 webAuthnManager = new WebAuthnManager(
	Arrays.asList(
		new PackedAttestationStatementValidator(),
	        new FIDOU2FAttestationStatementValidator(),
	        new TPMAttestationStatementValidator(),
	        new AndroidKeyAttestationStatementValidator(),
	        new AndroidSafetyNetAttestationStatementValidator(),
	        new AppleAnonymousAttestationStatementValidator(),
	        new NoneAttestationStatementValidator()
	),
	certPathTrustworthinessValidator(metadataStatementsProvider()),  //<-----
	new DefaultSelfAttestationTrustworthinessValidator()
	);
  webAuthnManager.getRegistrationDataValidator().getCustomRegistrationValidators().add(fidoMdsMetadataValidator());  <------
}
</code></pre>

  if you test with both enabled, "Metadata Service Tests" will not pass because of  "CertPathTrustworthinessValidator()".
  (It's probably not in the "metadata/test-tool" list.)
  

test spec&env
1. os : windows 10 pro
2. java : openjdk8
3. dev-tool : spring-boot 4.10v
4. fido-conformance-test-program version : 1.5.2v

**Comment by ynojima** [comment, marked as answer]
Thank you for using WebAuthn4J, and asking.

My FIDO conformance test target server is published as webauthn4j-spring-security sample app: https://github.com/webauthn4j/webauthn4j-spring-security/tree/master/samples/fido-server-conformance-test-app

Please check its bean definitions: https://github.com/webauthn4j/webauthn4j-spring-security/blob/master/samples/fido-server-conformance-test-app/src/main/java/com/webauthn4j/springframework/security/webauthn/sample/app/config/WebSecurityBeanConfig.java#L123-L145

When you run the test app, please update the URLs appropriately:
https://github.com/webauthn4j/webauthn4j-spring-security/blob/master/samples/fido-server-conformance-test-app/src/main/java/com/webauthn4j/springframework/security/webauthn/sample/app/config/WebSecurityBeanConfig.java#L183-L187
